CMakeLists.txt

```
@@ -1302,6 +1302,13 @@ check_symbol_exists(getrlimit      "${CURL_INCLUDES}" HAVE_GETRLIMIT)
 check_symbol_exists(setlocale      "${CURL_INCLUDES}" HAVE_SETLOCALE)
 check_symbol_exists(setmode        "${CURL_INCLUDES}" HAVE_SETMODE)
 check_symbol_exists(setrlimit      "${CURL_INCLUDES}" HAVE_SETRLIMIT)
+# On Windows we use SecureMemoryZero which is available in all supported
+# versions of Windows
+if(NOT WIN32)
+  check_symbol_exists(memset_s       "${CURL_INCLUDES};string.h" HAVE_MEMSET_S)
+  check_symbol_exists(explicit_bzero "${CURL_INCLUDES};string.h" HAVE_EXPLICIT_BZERO)
+  check_symbol_exists(explicit_memset "${CURL_INCLUDES};string.h" HAVE_EXPLICIT_BZERO)
```

```suggestion
  check_symbol_exists(memset_s "string.h" HAVE_MEMSET_S)
  if(NOT HAVE_MEMSET_S)
    check_symbol_exists(explicit_bzero "string.h" HAVE_EXPLICIT_BZERO)
    if(NOT HAVE_EXPLICIT_BZERO)
      check_symbol_exists(explicit_memset "string.h" HAVE_EXPLICIT_BZERO)
    endif()
  endif()
```
What do you think of making this a bit faster by stopping on the first hit?
Same would be useful for autotools, if there is a clean way.

---

That's not a bad idea, any way we can speed up the build is good. Added in the rebase to CMake for now.